src/pseudopeople/interface.py

```diff
@@ -64,7 +65,16 @@ def _generate_form(
             f"Source {source} must be either a pandas DataFrame or a path to a "
             "file containing a pandas DataFrame."
         )
-    return noise_form(form, data, configuration_tree, seed)
+
+    columns_to_keep = [c for c in form.columns]
+    for col in columns_to_keep:
+        data[col.name] = data[col.name].astype(col.dtype)
+        if col.dtype == "datetime64[ns]":
+            data[col.name] = data[col.name].apply(lambda x: x.date())
```

Comment:
Did you verify the actual `dtypes` of inputs? They should actually be correct for all columns except (ironically) the columns that are supposed to be strings. Those are currently categorical. It might be sufficient just to say:
```
if col.dtype == "str":
    data[col.name] = data[col.name].astype(str)
```

Reply:
Yeah, ok, I wasn't sure about that. I'll do this.

Reply:
@rmudambi 

this worries me in case we ever lose control of dtypes again. Is it time-consuming or otherwise dangerous to just cast dtypes even if they're already correct?

I can also do 
```
if col.dtype != data[col.name].dtype:
    data[col.name] = data[col.name].astype(col.dtype)
```

This might be the best of both worlds, though it will mean re-casting dtypes for state since the sample data makes up a fake state of "US"